### PR TITLE
[ShapeOpt] split code into header source

### LINKED ISSUE #9513: [ShapeOpt] Split ShapeOptimizationApplication code into header/source files
**Description**
ShapeOptimizationApplication should be split into header/source. This has been pointed out several times in the past by @philbucher

### PR DESCRIPTION
**Pull request by armingeiser**
**📝 Description**
Split the cpp code into header and source files.

No other changes have been made, only copied the function implementation to the .cpp file. 

closes #9513

### CODE REVIEW

**Comment by philbucher** [comment]
a) seems like you are missing some `KRATOS_API` 
b) how much fast does it compile now?

File: applications/ShapeOptimizationApplication/custom_utilities/filter_function.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    else if (FilterFunctionType == "linear")
+        mFilterFunctional =  [](double radius, double distance) {return std::max(0.0, (radius - distance) / radius);};
+
+    // Type 3: Constant function
+    else if (FilterFunctionType == "constant")
+        mFilterFunctional = [](double radius, double distance) {return 1.0;};
+
+    // Type 4: Cosine function
+    else if (FilterFunctionType == "cosine")
+        mFilterFunctional = [](double radius, double distance) {return std::max(0.0, 1-0.5*(1-cos(3.1415926535897932384626433832795/radius*distance)));};
```
Comment (philbucher) [inline applications/ShapeOptimizationApplication/custom_utilities/filter_function.cpp]:
```suggestion
        mFilterFunctional = [](double radius, double distance) {return std::max(0.0, 1-0.5*(1-std::cos(Globals::Pi/radius*distance)));};
```

**Comment by armingeiser** [comment]
>  b) how much fast does it compile now?

I tried to isolate the build of the ShapeOptApp.
Seems like its aktually 10s slower now (3 runs).

I did not expect a major improvement since all headers in the shapeopt app are included only once when exposing to python. But i did not expect a slow down...

**Comment by philbucher** [comment]
hm good question
I guess you don't have many cross-includes

Probably it will make a noticeable difference when you change one cpp file now
previously you would have to recompile a lot more, now not any more :)

File: applications/ShapeOptimizationApplication/custom_utilities/damping/damping_utilities.cpp [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+// Project includes
+// ------------------------------------------------------------------------------
+#include "custom_utilities/damping/damping_utilities.h"
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "spatial_containers/spatial_containers.h"
+#include "utilities/builtin_timer.h"
+#include "utilities/parallel_utilities.h"
+#include "custom_utilities/filter_function.h"
+#include "shape_optimization_application.h"
```
Comment (philbucher) [inline applications/ShapeOptimizationApplication/custom_utilities/damping/damping_utilities.cpp:29]: this is only necessary as the Variables are declared in this file
Another cleanup would be to move the vars to a separate file as is done for the other apps :)

**Review by philbucher — APPROVED** [review]
I didn't review but if it passes the CI I would say it is fine

I saw that there are some loops that could easily be parallelized, but this is for the future :)

File: applications/ShapeOptimizationApplication/custom_utilities/damping/damping_utilities.cpp (hunk 2 above)
Comment (armingeiser) [reply applications/ShapeOptimizationApplication/custom_utilities/damping/damping_utilities.cpp:29]: you mean importing the application? Yes, variables can be moved to another file in another PR =)

**Pull request merged**